services/localizely/localizely.service.js

```diff
@@ -0,0 +1,135 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService, InvalidResponse } = require('..')
+const { coveragePercentage } = require('../color-formatters')
+
+const keywords = [
+  'l10n',
+  'i18n',
+  'localization',
+  'internationalization',
+  'translation',
+  'translations',
+]
+
+const documentation = `
+  <p>
+    The read-only API token from the Localizely account is required to fetch necessary data.
+    <br/>
+    You can find more details regarding API tokens under <a href="https://app.localizely.com/account" target="_blank">My profile</a> page.
+  </p>
+  `
+
+const schema = Joi.object({
+  strings: Joi.number().required(),
+  reviewedProgress: Joi.number().required(),
+  languages: Joi.array()
+    .items(
+      Joi.object({
+        langCode: Joi.string().required(),
+        langName: Joi.string().required(),
+        strings: Joi.number().required(),
+        reviewed: Joi.number().required(),
+        reviewedProgress: Joi.number().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  token: Joi.string().required(),
+  languageCode: Joi.string().regex(/^[a-z]{2}(-[A-Z][a-z]{3})?(-[A-Z]{2})?$/),
+}).required()
+
+module.exports = class Localizely extends BaseJsonService {
+  static category = 'other'
+
+  static route = {
+    base: 'localizely/progress',
+    pattern: ':projectId/:branch*',
```

Comment:
Do you think there's a chance that one day localizely could make the progress data available anonymously, without requiring the token? 

We usually want [required parameters to be route params](https://github.com/badges/shields/blob/master/doc/badge-urls.md#badge-url-conventions) and at least today, the token is required. However, we have had a case before where a token was once required (and thus we required it as a route param) and then later on the service dropped the token requirement.

I think I could personally live with `token` being a (required) query param here if there is such a possibility, but if not then `token` should actually be a route param again.

cc @badges/shields-maintainers for input from others

Reply:
In general, with other services where the user can pass a token (e.g: circle, codecov, bitrise, poeditor, etc) the token is passed as a query param.

Reply:
Good point. The tokens for circle and codecov are optional (... _may_ provide) whereas the tokens are also **required** for bitrise and poeditor. I think keeping the token as a query param makes sense here as well then to be consistent, but I also think we should update our badge route documentation to make it clear that tokens are the exception to the rule:

> Parameters should always be part of the route if they are required to display a badge e.g: :packageName.

Should instead be something like

> Parameters should always be part of the route if they are required to display a badge e.g: :packageName, with the exception of tokens which should be provided as query parameters

Reply:
By the way, what does the token allow to do? Can it be used to expose some potentially sensitive data? I know we have a few existing badges that allow users to specify a token as a parameter, but we should probably be a bit careful about expanding the pattern any further. For reference, @paulmelnikow has advised against reusing the Circle/Codecov approach in https://github.com/badges/shields/issues/1495#issuecomment-370195236.

Reply:
Hi @calebcartwright, @chris48s and @PyvesB, 

The Localizely recently introduced the `read-only` API tokens for this case, so I believe that their API won't change.

Regarding security, with the `read-only` API tokens, users can access only to the translation status data and to download translations (already included in the project for localization). 
Having that in mind, I don't see any potential security risk.

Reply:
Thanks for the info @lzoran! Based on the information you've provided, I would say that this approach is fine in principle.

In practice, there's still opportunity for things to easily go wrong. I've just created a brand new Localizely account, and here's what I get in my profile section:
![Localizely](https://user-images.githubusercontent.com/10694593/92408933-4325b600-f13f-11ea-9f3e-948d2555b576.png)

Al token is automatically created for me and available by default, it has _write_ permissions as well. It's all too easy for users to not pay attention and simply press that copy button to use the default token, without realising that they're leaking write access to their project through the upload API.

At a minimum, I think we should make this explicit in the documentation: putting a clear warning telling users not to use the default token, and instead create a new one with only read permissions. Possibly an additional sentence briefly outlining the concrete security consequences of using a write token for their badge.

Reply:
You are right @PyvesB!
I've updated docs within [086c414](https://github.com/badges/shields/pull/5472/commits/086c41433e58a786b2ced608595a8baaa7de4957) commit.